File: cases/taint/csharp/return-relay-one-hop-positive/ReturnRelayOneHopPositive.cs
namespace DataFlowBench;

static class ReturnRelayOneHopPositive
{
    static int dfb_source() // DFB-SOURCE: return-one-hop-input
    {
        return 1;
    }

    static int Relay(int value) // DFB-WITNESS: return-one-hop-relay
    {
        return value;
    }

    static void dfb_sink(int value) { } // DFB-SINK: return-one-hop-sink

    static void Run()
    {
        int result = Relay(dfb_source());
        dfb_sink(result);
    }
}

File: cases/taint/c/return-relay-one-hop-positive/return_relay_one_hop_positive.c
int dfb_source(void) { // DFB-SOURCE: return-one-hop-input
    return 1;
}

int relay(int value) { // DFB-WITNESS: return-one-hop-relay
    return value;
}

void dfb_sink(int value) {} // DFB-SINK: return-one-hop-sink

void run(void) {
    int result = relay(dfb_source());
    dfb_sink(result);
}
